api/openapi.yaml

```diff
@@ -2350,6 +2554,107 @@ components:
 
+    Providers:
+      type: object
+      properties:
+        count:
+          type: integer
+          description: Total providers count according the given filters
+          readOnly: true
+        items:
+          type: array
+          description: List of providers in the given filters and page.
+            List length must be lower or equal to pageSize.
+          items:
+            $ref: '#/components/schemas/Provider'
+          readOnly: true
+
+    # NOTE Any changes made to Provider must be reflected in
+    # ProviderRelationship to ensure correct operation of the API when $expanding
+    Provider:
+      type: object
+      description: Describes an asset object.
+      properties:
+        id:
+          type: string
+        revision:
+          type: integer
+        displayName:
+          description: Human-readable name for the provider.
+          type: string
+        lastHeartbeatTime:
+          description: Timestamp of the last heartbeat from the provider.
+          type: string
+          format: date-time
+        status:
+          $ref: '#/components/schemas/ProviderStatus'
+        providerRuntimeVersion:
+          description: Version of the provider runtime that the provider was compiled with.
+          type: string
+        annotations:
+          $ref: '#/components/schemas/ProviderAnnotations'
+
+    ProviderRelationship:
+      type: object
+      description: Describes a relationship to a provider which can be expanded.
+      properties:
+        id:
+          type: string
+        revision:
+          type: integer
+          readOnly: true
+        displayName:
+          description: Human-readable name for the provider.
+          type: string
+          readOnly: true
+        lastHeartbeatTime:
+          description: Timestamp of the last heartbeat from the provider.
+          type: string
+          format: date-time
+          readOnly: true
+        status:
+          $ref: '#/components/schemas/ProviderStatus'
+          readOnly: true
+        providerRuntimeVersion:
+          description: Version of the provider runtime that the provider was compiled with.
+          type: string
+          readOnly: true
+        annotations:
+          $ref: '#/components/schemas/ProviderAnnotations'
+          readOnly: true
+
+    ProviderExists:
+      type: object
+      properties:
+        message:
+          description: Describes which unique constraint combination causes the conflict.
+          type: string
+          readOnly: true
+        provider:
+          $ref: '#/components/schemas/Provider'
+
+    ProviderStatus:
+      allOf:
+        - $ref: '#/components/schemas/CommonStatus'
+        - type: object
+          properties:
+            state:
+              description: Describes the health state of the provider.
+              type: string
+              enum:
+                - Healthy
+                - NotHealthy
+                - Unknown
+
+    ProviderAnnotations:
```

Comment:
Maybe just "Annotations", then we can reuse this schema to add annotations to other API objects in the future.